.github/workflows/ci.yml

```diff
@@ -410,34 +410,44 @@ jobs:
         with:
           path: up-cpp
 
-      - name: Run linters on source
-        id: source-linter
-        uses: cpp-linter/cpp-linter-action@v2
```

Comment:
This tool runs both clang-format and clang-tidy, and adds annotations to CI jobs / PR files. If possible, we should continue to use it for at least clang-format (and ideally clang-tidy too)

Reply:
Sounds good, I'll investigate the existing tool, vs installing clang-tidy-12 in CI.

The cpp-linter-action is will integrated with Github for adding annotations to CI jobs. It works well with clang-format, and we should continue using it. For clang-tidy, we might want to consider running clang-tidy separately if we need more detailed control or output. I will investigate both this tool, if this setup meets our goal then continuing using this tool would be much efficient and simpler.